lib/stat_tracker.rb

```diff
@@ -96,16 +88,42 @@ def worst_offense
     end
     team.team_name
   end
-
-  #season_stats
   
-  #Name of the Team with the most tackles in the season
-  # def most_tackles(season_id)
-    # find all GameTeam objects with the season id argument
-    # iterate through these GameTeam objects
-    # return a hash with the team id pointing to an array of all the tackle stats from each game
-    # 
+  def most_tackles(season_id)
+    season =  @games.find_all do |game|
+      game.season_id == season_id
+    end
+    season_game_ids = season.map do |game|
+      game.game_id
+    end
+
+    season_game_teams = []
+    @game_teams.each do |game_team|
+      if season_game_ids.include?(game_team.game_id)
+        season_game_teams << game_team
+      end
+    end
+    
+    game_team_tackles = {}
+    season_game_teams.each do |game_team|
+      game_team_tackles[game_team.team_id.to_sym] = []
+    end
 
+    season_game_teams.each do |game_team|
+      tackle_array = game_team_tackles[game_team.team_id.to_sym]
+      tackle_array << game_team.tackles.to_i
```

Comment:
I like the converting here as opposed to in the initialization as it allows the data to be manipulated easier.